lib/plug/router/utils.ex

```diff
@@ -100,6 +100,99 @@ defmodule Plug.Router.Utils do
     |> Enum.reject(&match?({"_" <> _var, _macro}, &1))
   end
 
+  @doc """
+  Removes format extensions from path as these are transformed into
+  guard clauses
+
+  ## Examples
+
+      iex> Plug.Router.Utils.remove_format_extension(":id.json")
+      ":id"
+
+      iex> Plug.Router.Utils.remove_format_extension(":id.js.map")
+      ":id"
+
+      iex> Plug.Router.Utils.remove_format_extension("foo-:id.json")
+      "foo-:id"
+  """
+  def remove_format_extension(segment) do
+    case Regex.run(~r/(.*):(.*?)\.(.*)$/, segment, capture: :all_but_first) do
+      nil -> segment
+      [buffer, identifier, _format] -> [buffer, ?:, identifier] |> IO.iodata_to_binary()
+    end
+  end
+
+  @doc """
+  Builds a list of path prefix, id, suffix info that can be used to
+  transform paths and generate guards for suffix identifier matching.
+
+  ## Examples
+
+      iex> Plug.Router.Utils.parse_suffix_identifier("/foo/:bar.json")
+      ["", "foo", {"", ":bar", ".json"}]
+
+      iex> Plug.Router.Utils.parse_suffix_identifier("/foo/:bar")
+      ["", "foo", ":bar"]
+
+      iex> Plug.Router.Utils.parse_suffix_identifier("/foo/:bar_baz")
+      ["", "foo", ":bar_baz"]
+
+      iex> Plug.Router.Utils.parse_suffix_identifier("/foo/:bar-json")
+      ["", "foo", {"", ":bar", "-json"}]
+
+      iex> Plug.Router.Utils.parse_suffix_identifier("/foo/:bar@example.com")
+      ["", "foo", {"", ":bar", "@example.com"}]
+
+      iex> Plug.Router.Utils.parse_suffix_identifier("/foo/:bar.js.map")
+      ["", "foo", {"", ":bar", ".js.map"}]
+
+      iex> Plug.Router.Utils.parse_suffix_identifier("/foo/bat-:bar.json")
+      ["", "foo", {"bat-", ":bar", ".json"}]
+
+      iex> Plug.Router.Utils.parse_suffix_identifier("/foo/bat-:id.app/baz/:bar.json")
+      ["", "foo", {"bat-", ":id", ".app"}, "baz", {"", ":bar", ".json"}]
+  """
+  def parse_suffix_identifier(path) when is_binary(path) do
+    String.split(path, "/")
```

Comment:
Use the `split` function in this module. :)

Reply:
But `Router.split` removes `""` for the leading `/` that is required when rejoining path after the transformation of suffix into guard clauses.